server/test/unit/com/thoughtworks/go/server/service/EntityHashingServiceTest.java

```diff
@@ -1,58 +1,66 @@
 package com.thoughtworks.go.server.service;
 
-import com.thoughtworks.go.config.CaseInsensitiveString;
+import com.thoughtworks.go.config.ConfigCache;
+import com.thoughtworks.go.config.MagicalGoConfigXmlWriter;
 import com.thoughtworks.go.config.PipelineConfig;
+import com.thoughtworks.go.config.registry.ConfigElementImplementationRegistry;
 import com.thoughtworks.go.helper.PipelineConfigMother;
 import com.thoughtworks.go.server.cache.GoCache;
-import com.thoughtworks.go.server.util.EntityDigest;
+import com.thoughtworks.go.util.CachedDigestUtils;
+import com.thoughtworks.go.util.ConfigElementImplementationRegistryMother;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Matchers.any;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyZeroInteractions;
 import static org.mockito.Mockito.when;
 
 public class EntityHashingServiceTest {
-    private EntityDigest entityDigest;
     private GoConfigService goConfigService;
     private GoCache goCache;
     private EntityHashingService entityHashingService;
+    private ConfigCache configCache;
+    private ConfigElementImplementationRegistry registry;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
 
     @Before
     public void setUp() {
-        this.entityDigest = mock(EntityDigest.class);
         this.goConfigService = mock(GoConfigService.class);
         this.goCache = mock(GoCache.class);
-        this.entityHashingService = new EntityHashingService(this.goConfigService, this.goCache);
+        this.configCache = new ConfigCache();
+        this.registry = ConfigElementImplementationRegistryMother.withNoPlugins();
+        this.entityHashingService = new EntityHashingService(this.goConfigService, this.goCache, configCache, registry);
 
-        this.entityHashingService.initializeWith(entityDigest);
     }
 
     @Test
-    public void shouldGenerateMD5ForAPipelineConfig() {
-        PipelineConfig pipelineConfig = mock(PipelineConfig.class);
-
-        when(goConfigService.pipelineConfigNamed(any(CaseInsensitiveString.class))).thenReturn(pipelineConfig);
-        when(entityDigest.md5ForPipeline(pipelineConfig)).thenReturn("pipeline_config_md5");
+    public void shouldThrowAnExceptionWhenObjectIsNull() {
+        thrown.expect(NullPointerException.class);
+        entityHashingService.md5ForEntity(null, null);
+    }
```

Comment:
Yes. Suppose there is no pipeline config named "foo". That was the intention of this test. But of course, the respective entity command would have already checked for such a case. Hopefully. :)